src/platforms/common/server/kms-utils/kms_connector.cpp

```diff
@@ -117,13 +130,34 @@ std::tuple<mgk::DRMModeCrtcUPtr, int> find_crtc_and_index_for_connector(
             {
                 if (encoder_supports_crtc_index(enc.get(), crtc_index))
                 {
+                    if (std::find(recent_crtc_ids.begin(), recent_crtc_ids.end(), crtc->crtc_id) != recent_crtc_ids.end())
+                    {
+                        skipped_crtcs.emplace_back(std::move(crtc), crtc_index);
+                        continue;
+                    }
+
+                    recent_crtc_ids.push_back(crtc->crtc_id);
                     return std::tuple<mgk::DRMModeCrtcUPtr, int>{std::move(crtc), crtc_index};
                 }
             }
         }
         crtc_index++;
     }
 
+    // We couldn't find a crtc we haven't used, so return the least recently returned one
+    for (auto pcrtc_id = recent_crtc_ids.begin(); pcrtc_id != recent_crtc_ids.end(); ++pcrtc_id)
+    {
+        for (auto& skipped_crtc : skipped_crtcs)
+        {
+            if (*pcrtc_id == get<0>(skipped_crtc)->crtc_id)
+            {
+                recent_crtc_ids.erase(pcrtc_id);
+                recent_crtc_ids.push_back(get<0>(skipped_crtc)->crtc_id);
```

Comment:
Should we also remove this from `skipped_crtcs`?

Reply:
No